rakelib/utils.rake

```diff
@@ -0,0 +1,58 @@
+# run a command, pipe output through 'xcpretty' and store the output in CI artifacts
+def xcpretty(cmd, task, subtask = '')
+  name = (task.name + (subtask.empty? ? '' : "_#{subtask}")).gsub(/[:-]/, "_")
+  command = [*cmd].join(' && ')
+  xcpretty = `which xcpretty`
+
+  if ENV['CI']
+    sh "set -o pipefail && (#{command}) | tee \"#{ENV['CIRCLE_ARTIFACTS']}/#{name}_raw.log\" | xcpretty --color --report junit --output \"#{ENV['CIRCLE_TEST_REPORTS']}/xcode/#{name}.xml\""
+  elsif xcpretty && $?.success?
+    sh "set -o pipefail && (#{command}) | xcpretty -c"
+  else
+    sh command
+  end
+end
+
+# run a command and store the output in CI artifacts
+def plain(cmd, task, subtask = '')
```

Comment:
We could even imagine doing the same for `xcrun`, turning it as a parameter like `xcrun: false` by default and using `xcrun: true` at call site would allow to inject the version select argument + `xcrun`

It would make the call site more clear as to when we want unfiltered output, when we want xcrun/version select, and when we want xcpretty, all under the same function.

Thoughts?